src/base/memory_monitor_sr.cpp

```diff
@@ -402,29 +485,116 @@ namespace cubperf
     m_mmon = mmon;
   }
 
-  int memory_monitor::aggregater::get_server_info (const MMON_SERVER_INFO &info)
+  void memory_monitor::aggregater::get_server_info (MMON_SERVER_INFO &info) const
   {
-    return 0;
+    strncpy (info.name, m_mmon->m_server_name.c_str (), m_mmon->m_server_name.size ());
+    info.total_mem_usage = m_mmon->m_total_mem_usage.load ();
+  }
+
+  int memory_monitor::aggregater::get_module_info (MMON_MODULE_INFO *&info, int module_index) const
+  {
+    int idx = 0;
+    int error = NO_ERROR;
+    if (module_index == -1)
+      {
+	// aggregate all detail memory information of modules
+	for (idx; idx < MMON_MODULE_LAST; idx++)
+	  {
+	    error = m_mmon->m_module[idx]->aggregate_stats (info[idx], false);
+	    if (error != NO_ERROR)
+	      {
+		return error;
```

Comment:
Q. I wonder it is really necessary to get all module information. What I was told is that a client will request several time even when it want know about all module. Then it's not needed.

If this is really needed, we should come up with a way to notify until which index it succeeds to aggregate.

Reply:
Server (memory monitoring manager) has to give detailed information of all registered modules when memmon gets the '-a' command from a user.  And due to the display specification of this utility, the atomicity of displaying information is limited by category, such as module, transaction, etc.

Reply:
I don't understand your answer. I mean, with `-a`, a client will request 3 times to get each module's information when there are 3 modules. I was told that it's the design since we should print as many as we can with even an error. And, this means we don't need for `get_module_info` to collect all module's information. That's my understanding.